Cannot link against a MSVC built Vc because it needs Vc_CDECL on some symbols

Vc version / revision | Operating System | Compiler & Version | Compiler Flags | Assembler & Version | CPU
----------------------|------------------|--------------------|----------------|---------------------|----
 1.4.2                     |      Windows            |         MSVC 19.29.30038.1           |       `/MD /O1 /Ob1 /DNDEBUG /fp:fast`         |   N/A                  | Ryzen 7 2700x

## Testcase

```cpp
#include <Vc/Vc>
#include <Vc/support.h>
#include <iostream>

int main() {
  if (Vc::isImplementationSupported(Vc::AVX2Impl)) {
    std::cout << "AVX2" << std::endl;
  } else if (Vc::isImplementationSupported(Vc::AVXImpl)) {
    std::cout << "AVX" << std::endl;
  } else if (Vc::isImplementationSupported(Vc::SSE41Impl)) {
    std::cout << "SSE 4.1" << std::endl;
  } else if (Vc::isImplementationSupported(Vc::SSSE3Impl)) {
    std::cout << "SSSE3" << std::endl;
  } else if (Vc::isImplementationSupported(Vc::SSE2Impl)) {
    std::cout << "SSE2" << std::endl;
  } else {
    std::cout << "Scalar" << std::endl;
  }

  Vc::float_v a;
  a[0] = rand();
  a[1] = rand();
  a[2] = rand();
  a[3] = rand();
  std::cout << a << std::endl;
  std::cout << a.sorted() << std::endl;
}
```

```cmake
cmake_minimum_required(VERSION 3.10)

project (test)

set(CMAKE_CXX_STANDARD 17)
set(CMAKE_CXX_STANDARD_REQUIRED ON)

find_package(Vc 1.3.0 REQUIRED)

add_executable(main main.cpp)

target_compile_options(main PRIVATE "/fp:fast")

target_link_libraries(main PUBLIC Vc::Vc)
```

## Actual Results

CMake finds and instantiates `Vc::Vc`, compilation succeeds, but linking fails:

```[proc] Executing command: "C:\Program Files\CMake\bin\cmake.EXE" --no-warn-unused-cli -DCMAKE_PREFIX_PATH:STRING=E:\VcDevel\install -DCMAKE_EXPORT_COMPILE_COMMANDS:BOOL=TRUE -DCMAKE_INSTALL_PREFIX:STRING=e:/VcDevelTest/install -He:/VcDevelTest/src -Be:/VcDevelTest/build -G "Visual Studio 16 2019" -T host=x64 -A x64
[cmake] Not searching for unused variables given on the command line.
[cmake] -- Selecting Windows SDK version 10.0.19041.0 to target Windows 10.0.19042.
[cmake] msvc_version.c
[cmake] c1: fatal error C1083: No se puede abrir el archivo origen: 'E:/VcDevel/install/lib/cmake/Vc/msvc_version.c': No such file or directory
[cmake] -- Detected Compiler: MSVC 
[cmake] -- Detected CPU: zen
[cmake] -- Configuring done
[cmake] -- Generating done
[cmake] -- Build files have been written to: E:/VcDevelTest/build
[main] Building folder: VcDevelTest 
[build] Starting build
[proc] Executing command: "C:\Program Files\CMake\bin\cmake.EXE" --build e:/VcDevelTest/build --config MinSizeRel --target ALL_BUILD -j 2 --
[build] Microsoft (R) Build Engine versión 16.10.2+857e5a733 para .NET Framework
[build] Copyright (C) Microsoft Corporation. Todos los derechos reservados.
[build] 
[build]   main.cpp
[build] E:\VcDevel\install\include\Vc\sse\detail.h(679,15): warning C4244: 'return': conversión de 'short' a 'Vc_1::schar'; posible pérdida de datos [E:\VcDevelTest\build\main.vcxproj]
[build] E:\VcDevel\install\include\Vc\avx\detail.h(737,48): warning C4146: operador unario menos aplicado a un tipo unsigned; el resultado aún no tiene signo [E:\VcDevelTest\build\main.vcxproj]
[build] E:\VcDevel\install\include\Vc\avx\detail.h(738,48): warning C4146: operador unario menos aplicado a un tipo unsigned; el resultado aún no tiene signo [E:\VcDevelTest\build\main.vcxproj]
[build] E:\VcDevel\install\include\Vc\avx/vector.h(324,53): warning C4244: '=': conversión de 'U' a 'float'; posible pérdida de datos [E:\VcDevelTest\build\main.vcxproj]
[build]           with
[build]           [
[build]               U=int
[build]           ]
[build] E:\VcDevel\install\include\Vc\avx/vector.h(323): message : al compilar la función del miembro clase plantilla "void Vc_1::Vector<float,Vc_1::VectorAbi::Avx>::set<ValueType>(Vc_1::Vector<float,Vc_1::VectorAbi::Avx> &,int,U &&) noexcept(<expr>)" [E:\VcDevelTest\build\main.vcxproj]
[build]           with
[build]           [
[build]               ValueType=int,
[build]               U=int
[build]           ]
[build] E:\VcDevel\install\include\Vc\common\elementreference.h(79): message : al compilar la función del miembro clase plantilla "Vc_1::Detail::ElementReference<Vc_1::Vector<float,Vc_1::VectorAbi::Avx>,U> &Vc_1::Detail::ElementReference<U,U>::operator =<int>(T &&) noexcept(<expr>) &&" [E:\VcDevelTest\build\main.vcxproj]
[build]           with
[build]           [
[build]               U=Vc_1::Vector<float,Vc_1::VectorAbi::Avx>,
[build]               T=int
[build]           ]
[build] E:\VcDevel\install\include\Vc\avx/vector.h(326,31): warning C4244: 'argumento': conversión de 'U' a 'float'; posible pérdida de datos [E:\VcDevelTest\build\main.vcxproj]
[build]           with
[build]           [
[build]               U=int
[build]           ]
[build] E:\VcDevel\install\include\Vc\common\elementreference.h(82): message : Vea la referencia a la creación de una instancia de la función plantilla "void Vc_1::Vector<float,Vc_1::VectorAbi::Avx>::set<ValueType>(Vc_1::Vector<float,Vc_1::VectorAbi::Avx> &,int,U &&) noexcept" que se está compilando [E:\VcDevelTest\build\main.vcxproj]
[build]           with
[build]           [
[build]               ValueType=int,
[build]               U=int
[build]           ]
[build] E:\VcDevel\install\include\Vc\common\elementreference.h(82): message : Vea la referencia a la creación de una instancia de la función plantilla "void Vc_1::Vector<float,Vc_1::VectorAbi::Avx>::set<ValueType>(Vc_1::Vector<float,Vc_1::VectorAbi::Avx> &,int,U &&) noexcept" que se está compilando [E:\VcDevelTest\build\main.vcxproj]
[build]           with
[build]           [
[build]               ValueType=int,
[build]               U=int
[build]           ]
[build] E:\VcDevelTest\src\main.cpp(23): message : Vea la referencia a la creación de una instancia de la función plantilla "Vc_1::Detail::ElementReference<Vc_1::Vector<float,Vc_1::VectorAbi::Avx>,U> &Vc_1::Detail::ElementReference<U,U>::operator =<int>(T &&) noexcept &&" que se está compilando [E:\VcDevelTest\build\main.vcxproj]
[build]           with
[build]           [
[build]               U=Vc_1::Vector<float,Vc_1::VectorAbi::Avx>,
[build]               T=int
[build]           ]
[build] E:\VcDevelTest\src\main.cpp(23): message : Vea la referencia a la creación de una instancia de la función plantilla "Vc_1::Detail::ElementReference<Vc_1::Vector<float,Vc_1::VectorAbi::Avx>,U> &Vc_1::Detail::ElementReference<U,U>::operator =<int>(T &&) noexcept &&" que se está compilando [E:\VcDevelTest\build\main.vcxproj]
[build]           with
[build]           [
[build]               U=Vc_1::Vector<float,Vc_1::VectorAbi::Avx>,
[build]               T=int
[build]           ]
[build] main.obj : error LNK2019: símbolo externo "public: static void __cdecl Vc_1::CpuId::init(void)" (?init@CpuId@Vc_1@@SAXXZ) sin resolver al que se hace referencia en la función "void __cdecl `dynamic initializer for 'public: static struct Vc_1::detail::RunCpuIdInit<0> Vc_1::detail::RunCpuIdInit<0>::tmp''(void)" (??__E?tmp@?$RunCpuIdInit@$0A@@detail@Vc_1@@2U123@A@@YAXXZ) [E:\VcDevelTest\build\main.vcxproj]
[build] main.obj : error LNK2019: símbolo externo "bool __cdecl Vc_1::isImplementationSupported(enum Vc_1::Implementation)" (?isImplementationSupported@Vc_1@@YA_NW4Implementation@1@@Z) sin resolver al que se hace referencia en la función main [E:\VcDevelTest\build\main.vcxproj]
[build] main.obj : error LNK2019: símbolo externo "class Vc_1::Vector<float,struct Vc_1::VectorAbi::Avx> __cdecl Vc_1::Detail::sorted<7,float,struct Vc_1::enable_if_default_type>(class Vc_1::Vector<float,struct Vc_1::VectorAbi::Avx>)" (??$sorted@$06MUenable_if_default_type@Vc_1@@@Detail@Vc_1@@YA?AV?$Vector@MUAvx@VectorAbi@Vc_1@@@1@V21@@Z) sin resolver al que se hace referencia en la función main [E:\VcDevelTest\build\main.vcxproj]
[build] E:\VcDevelTest\build\MinSizeRel\main.exe : fatal error LNK1120: 3 externos sin resolver [E:\VcDevelTest\build\main.vcxproj]
[build] Build finished with exit code 1
```

## Expected Results

Compilation and linking successful with no errors.

--- a/Vc/avx/detail.h
+++ b/Vc/avx/detail.h
@@ -937,7 +937,7 @@ Vc_INTRINSIC __m256i avx_broadcast( uchar x) { return _mm256_set1_epi8(x); }
 // sorted{{{1
 template <Vc::Implementation Impl, typename T,
           typename = enable_if<(Impl >= AVXImpl && Impl <= AVX2Impl)>>
-Vc_CONST_L AVX2::Vector<T> sorted(AVX2::Vector<T> x) Vc_CONST_R;
+Vc_CONST_L AVX2::Vector<T> Vc_VDECL sorted(AVX2::Vector<T> x) Vc_CONST_R;
 template <typename T> Vc_INTRINSIC Vc_CONST AVX2::Vector<T> sorted(AVX2::Vector<T> x)
 {
     return sorted<CurrentImplementation::current()>(x);

--- a/Vc/cpuid.h
+++ b/Vc/cpuid.h
@@ -67,7 +67,7 @@ class CpuId
          * Will be executed automatically before main, but not necessarily before other functions
          * executing before main.
          */
-        static void init();
+        static void Vc_VDECL init();
 
         //! Return the cache line size in bits.
         static inline ushort cacheLineSize() { return static_cast<ushort>(s_cacheLineSize) * 8u; }

--- a/Vc/sse/detail.h
+++ b/Vc/sse/detail.h
@@ -772,7 +772,7 @@ Vc_INTRINSIC  uchar max(__m128i a,  uchar) {
 
 // sorted{{{1
 template <Vc::Implementation, typename T>
-Vc_CONST_L SSE::Vector<T> sorted(SSE::Vector<T> x) Vc_CONST_R;
+Vc_CONST_L SSE::Vector<T> Vc_VDECL sorted(SSE::Vector<T> x) Vc_CONST_R;
 template <typename T> Vc_INTRINSIC Vc_CONST SSE::Vector<T> sorted(SSE::Vector<T> x)
 {
     static_assert(!CurrentImplementation::is(ScalarImpl),

--- a/Vc/support.h
+++ b/Vc/support.h
@@ -72,8 +72,7 @@ unsigned int extraInstructionsSupported();
  *
  * \param impl The SIMD target to test for.
  */
-Vc_TARGET_NO_SIMD
-bool isImplementationSupported(Vc::Implementation impl);
+Vc_TARGET_NO_SIMD bool Vc_VDECL isImplementationSupported(Vc::Implementation impl);
 
 /**
  * \internal

